### ISSUE TITLE
Reflect on design-patterns to combat complexities with having multiple input modalities

### ISSUE THREAD

**Issue by zepumph** [phetsims/ratio-and-proportion#380]
It would be nice to get a subgroup discussion with devs to discuss some of the challenges that came about from Ratio and Proportion. 

The primary challenges from Ratio and Proportion are as follows:
* Strategies for balancing modularity with the need for Interactive Description (mostly natural language alert sentences) to potentially know about many different aspects of the view and model in a given space. 
  * For example (I think there are better ones too) https://github.com/phetsims/ratio-and-proportion/blob/f905c93a750331f6a5cb3f9c801aba5539b59a15/js/common/view/BothHandsPDOMNode.js#L162-L168, an option instead of reaching into the inputListener for the description.
* Challenge with using common code components when you need to know about when input occurs, instead of when the Property changes (which is often too late or too general for the description needed)
  * For example: https://github.com/phetsims/ratio-and-proportion/blob/f905c93a750331f6a5cb3f9c801aba5539b59a15/js/create/view/CreateScreenView.js#L74-L81, which needs to know about when you have input on a checkbox, not when the Property changes.

**Comment by chrisklus** [comment]
From 5/6/21 dev meeting:

Subgroup volunteers: @samreid, @jessegreenberg, @jonathanolson
Thanks!

**Comment by zepumph** [comment]
I don't think that @samreid and my schedules with align next week. I'll reach out about a potential time the week of 5/17

**Comment by zepumph** [comment]
Tagging @emily-phet as this is a quarterly goal to discuss.

**Comment by zepumph** [comment]
@jonathanolson, @jessegreenberg, @samreid, and @zepumph discussed this today. We took a general tour of interactive description in Ratio and Proportion, as well as the "describer" pattern. In general that pattern means constructing `*Describer` types  high up in the view (like in the general ScreenView class) with all of the model data they may need, and then passing those accordingly down the scene graph, often to quite specific Nodes. This is because interactive description is most often passed through to the exact Nodes rendered in the PDOM. Something we didn't discuss is how that differs from voicing, where we can put many of the responses on any Node for speaking.

We then talked about the second example in the first comment above, and created https://github.com/phetsims/sun/issues/701 as a general solution to exposing when input occurs on common code, instead of just the Property changing. 

Overall @jonathanolson and @samreid didn't have large, general concerns about our describer pattern, which was good to confirm. That said we also recognized that the only true way to get a full review of interactive description as a feature is to have devs integrate it themselves into a sim. This will happen more and more with the team (for @samreid with EEE). 

For now I'm going to close this issue, since the discussion occurred.

**Comment by emily-phet** [comment]
Just wanted to add a comment. From my read of the summary above, it seems like group discussion centered on interactive description. I just want to note that to me, the issue to tackle spans beyond interactive description, to ultimately develop strategies for designing sim models that work well across different input and output modalities. Interactive description is one complex subset of the bigger challenge.
